extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryPresenterCompilerStates.java

```diff
@@ -96,10 +96,38 @@
   }
 
   /**
+   * Compiler state to compile for the generic collection container type of the associated
+   * {@link Repository}.
+   */
+  interface RPCollectionCompile<TVal> {
+    /**
+     * Creates a {@link RepositoryPresenter} for a @{link Repository} of a type that can be
+     * converted to a {@link List} of items using the {@code converter}.
+     */
+    @NonNull
+    <TCol> RepositoryPresenter<TCol> forCollection(@NonNull Function<TCol, List<TVal>> converter);
+  }
+
+  /**
+   * Compiler state to compile for the generic collection container type of the associated
+   * {@link Repository}.
+   */
+  interface RPTypedCollectionCompile<TVal, TCol> {
+    /**
+     * Creates a {@link RepositoryPresenter} for a @{link Repository} of a type that can be
+     * converted to a {@link List} of items using the {@code converter}.
+     */
+    @NonNull
+    <TColE extends TCol> RepositoryPresenter<TColE> forCollection(
+        @NonNull Function<TColE, List<TVal>> converter);
+  }
+
+  /**
    * Compiler state to specify how to bind the {@link Repository} item to the view inflated by the
    * layout.
    */
-  interface RPMain<T> extends RPItemCompile<T>, RPSpecificCollectionCompile<T> {
+  interface RPMain<T> extends RPItemCompile<T>,
+      RPSpecificCollectionCompile<T>, RPCollectionCompile<T> {
```

Comment:
it seems to me that `RPSpecificCollectionCompile` and `RPCollectionCompile` are not used as a return type and can therefore be inlined into `RPMain`.